### PR TITLE
Display Google OAuth status on Accounts page (Auth Method column)

### PR DESCRIPTION
**Pull request by leonj1**
## Summary
- Implement a visual indicator for authentication method (OAuth vs IMAP) directly on the Accounts page.
- Add an optional auth_method field to the account API response to support a single fetch for status data.
- Introduce an Auth Method column in the accounts table and render per-account badges indicating the authentication method (OAuth Connected for OAuth, IMAP for IMAP, Not Configured for unknown/null).
- Include a new draft spec and associated tests/docs to guide and verify the OAuth status badge design and data flow.

### Backend / API
- Extend EmailAccountInfo API response model to include an optional auth_method field (e.g., 'oauth' or 'imap') so /api/accounts can render badges without extra API calls.
- Update GET /api/accounts response mapping to include account.auth_method.

### Frontend
- Add a new column "Auth Method" to the Accounts page table (placed after Email Address).
- Render a badge for each account based on account.auth_method:
  - oauth  -> green badge with text "OAuth Connected"
  - imap   -> gray badge with text "IMAP"
  - unknown/NULL -> neutral badge (e.g., "Not Configured")
- Implement badge rendering in accounts template and JavaScript rendering logic.
- Add CSS styles for the new badge variants and ensure accessibility attributes (aria-label) are present on badges.

### Data Model & Architecture
- EmailAccountInfo carries optional auth_method bound to the database's auth_method column.
- Frontend uses the new field to display the appropriate badge without extra API calls.

### Documentation & Specs
- Added a new draft spec: specs/DRAFT-oauth-status-badge.md outlining the OAuth status badge design and data flow.

### Files touched (high level)
- Backend: models/account_models.py (extend EmailAccountInfo with auth_method)
- Backend: api_service.py (include auth_method in GET /api/accounts response)
- Frontend: frontend/templates/accounts.html (add column and badge rendering)
- Specs: specs/DRAFT-oauth-status-badge.md (new draft spec)

## Testing plan
- API test: GET /api/accounts returns auth_method for each account (oauth, imap, unknown/NULL).
- Frontend/UI test: Accounts table includes a new Auth Method column with correct badges for various accounts.
- Edge cases: NULL or unknown auth_method shows a neutral/Not Configured badge; mixed accounts render appropriate badges in the same table.

### Manual test steps
1. Load the Accounts page and verify there is a new Auth Method column.
2. For accounts configured with OAuth, confirm the badge shows "OAuth Connected" with a green appearance.
3. For accounts using IMAP, confirm the badge shows "IMAP" with a gray appearance.
4. For accounts with NULL/unknown auth_method, confirm a neutral "Not Configured" badge is shown.
5. Ensure other columns and table layout remain correct.

## Notes
- This change relies on the existing auth_method column already present in the EmailAccount model.
- The implementation favors a single API call by including auth_method in the accounts response, avoiding per-account status calls.

## References / Task
- Task: https://www.terragonlabs.com/task/9a09bb6c-e7d7-4d92-b427-7cd48b79b1e8

---
Generated by the updated patch set and aligned with OAuth badge design and data flow expectations.

## Summary by CodeRabbit

* **New Features**
  * Accounts page adds an "Auth Method" column with badges: green "OAuth Connected", gray "IMAP", or "Not Configured".
* **Accessibility**
  * Badges and column header include aria attributes for screen readers.
* **Performance**
  * Accounts list loads more efficiently by returning auth method with account data to avoid redundant lookups.
* **Tests**
  * Added comprehensive tests covering badge display, API responses, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: clients/account_category_client.py [hunk 1]
```diff
@@ -137,10 +137,10 @@ def _validate_email_address(self, email_address: str) -> str:
...
         self,
         email_address: str,
-        display_name: Optional[str] = None,
-        app_password: Optional[str] = None,
-        auth_method: str = "imap",
-        oauth_refresh_token: Optional[str] = None,
+        display_name: Optional[str],
+        app_password: Optional[str],
+        auth_method: Optional[str],
+        oauth_refresh_token: Optional[str],
```
[the remark this answers is not in the document]
Comment (leonj1) [reply clients/account_category_client.py]: @terragon-labs

File: clients/account_category_client_interface.py [hunk 2]
```diff
@@ -14,13 +14,15 @@ class AccountCategoryClientInterface(ABC):
     """Interface for managing email accounts and category statistics."""
 
     @abstractmethod
-    def get_or_create_account(self, email_address: str, display_name: Optional[str] = None) -> EmailAccount:
+    def get_or_create_account(self, email_address: str, display_name: Optional[str], app_password: Optional[str], auth_method: Optional[str]) -> EmailAccount:
```
[the remark this answers is not in the document]
Comment (leonj1) [reply clients/account_category_client_interface.py]: @terragon-labs

File: frontend/templates/accounts.html [hunk 3]
```diff
@@ -514,6 +544,30 @@ <h5 class="modal-title">Confirm Deletion</h5>
...
             emailCell.appendChild(emailSpan);
 
+            // Create Auth Method cell
+            const authMethodCell = document.createElement('td');
+            authMethodCell.setAttribute('data-label', 'Auth Method');
+            const authBadge = document.createElement('span');
+            authBadge.className = 'auth-badge';
+
+            // Determine badge text and style based on auth_method (case-insensitive)
+            const authMethod = account.auth_method ? account.auth_method.toLowerCase() : null;
```
[the remark this answers is not in the document]
Comment (leonj1) [reply frontend/templates/accounts.html]: @terragon-labs

File: clients/account_category_client.py [hunk 4]
```diff
@@ -530,7 +530,7 @@ def _get_top_categories_impl(self, session: Session, account_id: int, email_addr
             logger.error(f"Unexpected error in _get_top_categories_impl: {str(e)}")
             raise
     
-    def get_all_accounts(self, active_only: bool = True) -> List[EmailAccount]:
+    def get_all_accounts(self, active_only: bool) -> List[EmailAccount]:
```
[the remark this answers is not in the document]
Comment (leonj1) [reply clients/account_category_client.py]: @terragon-labs

File: tests/test_account_category_service.py [hunk 5]
```diff
@@ -94,12 +94,12 @@ def test_ensure_no_get_account_method(self):
     def test_create_or_update_account(self):
         """Test get_or_create_account method."""
         # First creation
-        account1 = self.service.get_or_create_account(self.test_email)
+        account1 = self.service.get_or_create_account(self.test_email, None, None, None)
```
[the remark this answers is not in the document]
Comment (leonj1) [reply tests/test_account_category_service.py]: @terragon-labs

File: tests/test_account_email_processor_service.py [hunk 6]
```diff
@@ -141,7 +141,7 @@ def test_process_account_not_found(self):
     def test_process_account_no_password(self):
         """Test process_account when account has no app password configured."""
         # Setup - create account without app password
-        account = self.account_category_client.get_or_create_account(self.test_email)
+        account = self.account_category_client.get_or_create_account(self.test_email, None, None, None)
```
[the remark this answers is not in the document]
Comment (leonj1) [reply tests/test_account_email_processor_service.py]: @terragon-labs

File: tests/test_email_account_app_password_integration.py [hunk 7]
```diff
@@ -90,7 +90,7 @@ def test_email_account_has_app_password_attribute(self):
         test_password = self.fake.password()
 
         # Create account using real client
-        account = self.real_client.get_or_create_account(test_email)
+        account = self.real_client.get_or_create_account(test_email, None, None, None)
```
[the remark this answers is not in the document]
Comment (leonj1) [reply tests/test_email_account_app_password_integration.py]: @terragon-labs

File: tests/test_email_account_app_password_integration.py [hunk 8]
```diff
@@ -146,7 +146,7 @@ def test_account_without_app_password_returns_error(self):
         api_token = self.fake.uuid4()
 
         # Create account WITHOUT app_password
-        account = self.real_client.get_or_create_account(test_email)
+        account = self.real_client.get_or_create_account(test_email, None, None, None)
```
[the remark this answers is not in the document]
Comment (leonj1) [reply tests/test_email_account_app_password_integration.py]: @terragon-labs
